cilium-health/launch/endpoint.go

```diff
@@ -228,12 +229,20 @@ type EndpointAdder interface {
 }
 
 // LaunchAsEndpoint launches the cilium-health agent in a nested network
-// namespace and attaches it to Cilium the same way as any other endpoint,
-// but with special reserved labels.
+// namespace and attaches it to Cilium the same way as any other endpoint, but
+// with special reserved labels.
 //
 // CleanupEndpoint() must be called before calling LaunchAsEndpoint() to ensure
 // cleanup of prior cilium-health endpoint instances.
-func LaunchAsEndpoint(baseCtx context.Context, owner regeneration.Owner, n *node.Node, mtuConfig mtu.Configuration, epMgr EndpointAdder, proxy endpoint.EndpointProxy, allocator cache.IdentityAllocator) (*Client, error) {
+func LaunchAsEndpoint(baseCtx context.Context,
+	owner regeneration.Owner,
+	n *node.Node,
+	mtuConfig mtu.Configuration,
+	epMgr EndpointAdder,
+	proxy endpoint.EndpointProxy,
+	allocator cache.IdentityAllocator,
+	eniHealthRouting *enirouting.RoutingInfo) (*Client, error) {
```

Comment:
As a general cleanup, we can avoid directly depending on the eni package from this health endpoint package by defining an interface something like the following, then accepting that as a parameter into this function:

```
type routeConfiguration interface{
    ConfigureRoutes(ip string, mtu int, masquerade bool)
}
```

The other step of this would require instantiating an object to associate this function with, so basically adjust the `Install()` function below to take the `RoutingInfo` as a pointer receiver.

I'd expect this interface to be `nil` if ENI is disabled, or an instance of the `RoutingInfo` if ENI is enabled.

In this specific case I don't think it makes a huge difference, but over time this general approach helps to separate the code, helps unit testing, and avoids tying everything together too tightly with direct implementations.

Reply:
Thanks, this is good feedback. I'll try this out.